common/lib/common-utils/src/test/mocha/idleTaskScheduler.spec.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { SinonFakeTimers, useFakeTimers } from "sinon";
+import * as idleTask from "../../idleTaskScheduler";
+
+describe("Idle task scheduler", () => {
+    let clock: SinonFakeTimers;
+
+    before(() => {
+        clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+        clock.reset();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    function someTask(x: number): boolean {
+        return x > 3 ? true : false;
+    }
+
+    it("Should schedule and run a synchronous task during idle time", async () => {
+        const promise = idleTask.scheduleIdleTask(() => {
+            return someTask(5);
+        }, 1000);
+
+        clock.tick(1100);
+        return promise.then((result) => assert(result));
```

Comment:
I do not know if mocha will raise errors (and fail tests) when it happens in continuation outside of test code (this method will return before promise chain completes).
You are better off making this test async, and then using await inside of it, to ensure that mocha will wait for the end of the result.